models/weecology-esss.R

```diff
@@ -0,0 +1,155 @@
+    # prep the aic data tabe
+      
+      aic <- data.frame(date = as.Date(forecast_date), 
+                        currency = 'abundance', 
+                        model = 'Weecology-ESSS', 
+                        level = level, species = 'total', 
+                        aic = as.numeric(etsmf$model$aic), 
+                        fit_start_newmoon = min(abundances$newmoonnumber),
+                        fit_end_newmoon = max(abundances$newmoonnumber), 
+                        initial_newmoon = max(abundances$newmoonnumber))
+
+    # return the output
+
+      return(list(fdt, aic))
+  }
+
+
+
+
+# Run model on all plots and just controls
+
+  # Get data
+
+    all <- read.csv("data/rodent_all.csv")
+    controls <- read.csv("data/rodent_controls.csv")
+    model_metadata <- yaml.load_file("data/model_metadata.yaml")
+    forecast_date <- as.Date(model_metadata$forecast_date)
+    filename_suffix <- model_metadata$filename_suffix
+    forecast_months <- model_metadata$forecast_months
+    forecast_years <- model_metadata$forecast_years
+    forecast_newmoons <- model_metadata$forecast_newmoons
+
+  # Forecast All plots
+
+    allresults <- weecology.esss(abundances = all, 
+                                 forecast_date = forecast_date,
+                                 forecast_months = forecast_months, 
+                                 forecast_years = forecast_years,
+                                 forecast_newmoons = forecast_newmoons,
+                                 level = "All",
+                                 num_forecast_months = 12, CI_level = 0.9)
+
+  # Forecast Control plots
+
+    controlsresults <- weecology.esss(abundances = controls, 
+                                      forecast_date = forecast_date,
+                                      forecast_months = forecast_months, 
+                                      forecast_years = forecast_years,
+                                      forecast_newmoons = forecast_newmoons,
+                                      level = "Controls",
+                                      num_forecast_months = 12, 
+                                      CI_level = 0.9)
+
+  # Combine output
+  #  warnings are suppressed here because they're just associated with 
+  #  coercing un-matching sets of factors to character vectors to bind
+
+    forecasts <- suppressWarnings(bind_rows(allresults[1], 
+                                            controlsresults[1]))
+    forecast_aics <- suppressWarnings(bind_rows(allresults[2], 
+                                                controlsresults[2]))
+
+# Write results out
+
+  write.csv(forecasts, 
+            file.path('tmp', 
+                paste("Weecology-ESSS", filename_suffix, ".csv", sep = "")),
+            row.names = FALSE)
+  write.csv(forecast_aics, 
+            file.path('tmp', 
+                paste("Weecology-ESSS", filename_suffix, "_model_aic.csv", 
+                      sep = "")),
+            row.names = FALSE)
```

Comment:
This sort of multi-level indenting for line length can make code difficult to read. In this case I think that pulling the filename creation out might make it more readable, e.g.,

```
fcast_path <- file.path('tmp', paste("Weecology-ESSS", filename_suffix, ".csv", sep = ""))
write.csv(forecasts, fcast_path, row.names = FALSE)
```

Once the "Weecology-" is pulled this fits in 80 chars without the challenging read.